File: 57.buscar_numero_en_lista/buscar_numero.py
""" Sin funciones
print("-----------------------------")
print("- BUSCAR NUMERO EN CONJUNTO -")
print("-----------------------------")

b = int(input("Número a buscar: "))

a = [1, 2, 3, 4, 5]
r = False

for i in a:
    if i == b:
        print("Lo encontre")
        r = True
if r == False:
    print("No lo encontre")
print("\nEso era")
"""
def buscar_dato_en_lista(dato_a_buscar, lista):
    r = False
    for i in lista:
        if i == dato_a_buscar:
            r = True
    return r

dato = int(input("Número a buscar: "))
lista = [1, 2, 3, 4, 5]

if buscar_dato_en_lista(dato, lista):
    print("Lo encontre")
else:
    print("No lo encontre")

File: 57.buscar_numero_en_lista/BuscarNumero.java
import javax.swing.JOptionPane;

public class BuscarNumero{
  //Processing
  public static boolean buscarDatos(int dato, int[] catalogo){
    boolean encontrado = false;
    
    for(int i = 0; i<catalogo.length; i++){
      if(catalogo[i] == dato){
	encontrado = true;
      }
    }
    return encontrado;
  }

  public static void main(String[] args){
    int[] lista = {1, 2, 3, 4, 5};
    //Input
    int numero = Integer.parseInt(JOptionPane.showInputDialog("Ingrese el número que quiere buscar"));
    //Processing
    if(buscarDatos(numero, lista)){
      JOptionPane.showMessageDialog(null, "Lo hallé");

    }
    else{
      JOptionPane.showMessageDialog(null, "No lo hallé");
    }

    System.exit(0);
  }
}
